app/models/stacks_media_stream.rb

```diff
@@ -0,0 +1,65 @@
+##
+# media stream via stacks
+class StacksMediaStream < StacksFile
+  TYPE_TO_MANIFEST_FILE = {
+    hls: 'playlist.m3u8',
+    dash: 'manifest.mpd'
+  }.freeze
+
+  attr_accessor :format
+
+  def to_playlist_url
+    streaming_url_for(:hls)
+  end
+
+  def to_manifest_url
+    streaming_url_for(:dash)
+  end
+
+  private
+
+  def streaming_url_for(type)
+    return unless file_name && streaming_file_prefix
+    suffix = TYPE_TO_MANIFEST_FILE[type]
```

Comment:
manifest_filename  would be slightly more generic, imo, than "playlist".    it is being assigned from something called TYPE_TO_MANIFEST_FILE ...